.github/workflows/codeql-analysis.yml

```diff
@@ -1,76 +1,12 @@
-# For most projects, this workflow file will not need changing; you simply need
-# to commit it to your repository.
-#
-# You may wish to alter this file to override the set of languages analyzed,
-# or to provide custom queries or build logic.
-#
-# ******** NOTE ********
-# We have attempted to detect the languages in your repository. Please check
-# the `language` matrix defined below to confirm you have the correct set of
-# supported CodeQL languages.
-#
-name: "CodeQL"
-
+name: 'CodeQL'
 on:
-  push:
-    branches: [ main ]
   pull_request:
-    # The branches below must be a subset of the branches above
-    branches: [ main ]
+    types: [ opened, reopened, synchronize ]
+    paths:
+      - 'src/main/java/**'
   schedule:
-    - cron: '22 2 * * 5'
+    - cron: '0 2 * * 0'
```

Comment:
Maybe it would be better to do this in-hours so that people don't get out-of-hours notifications?

Reply:
0 8 * * 1-5 ?